src/host/ncp_host.cpp

```diff
@@ -307,12 +307,14 @@ void NcpHost::Update(MainloopContext &aMainloop)
     mCliDaemon.UpdateFdSet(aMainloop);
 }
 
-#if OTBR_ENABLE_SRP_ADVERTISING_PROXY
+#if OTBR_ENABLE_MDNS
 void NcpHost::SetMdnsPublisher(Mdns::Publisher *aPublisher)
 {
     mNcpSpinel.SetMdnsPublisher(aPublisher);
 }
+#endif
 
+#if OTBR_ENABLE_SRP_ADVERTISING_PROXY
```

Comment:
I suspect this could be improved but I'm not sure due to the lack of context. Maybe we can improve it in a future PR.

Reply:
Agree. Using `OTBR_ENABLE_MDNS` looks good to me. We can further add checks  that when `OTBR_ENABLE_SRP_ADVERTISING_PROXY` or `OTBR_ENABLE_BORDER_AGENT` is enabled, `OTBR_ENABLE_MDNS` must be enabled.